### DISCUSSION TITLE
SOLVED: Problem upgrading

### DISCUSSION THREAD

**Discussion by sollermun** [kimai/kimai#3889] [On-Premise: Installation & Updates]
Hi, I have been using Kimai 2 for a while and I decided to try updating by following the instructions here:  https://www.kimai.org/documentation/updates.html

When I do so, I am running into the following error when I run the command "composer install --optimize-autoloader -n"

PHP Parse error:  syntax error, unexpected 'Lock' (T_STRING), expecting function (T_FUNCTION) or const (T_CONST) in [directory structure]/kimai2/vendor/symfony/flex/src/Command/RecipesCommand.php on line 35

Parse error: syntax error, unexpected 'Lock' (T_STRING), expecting function (T_FUNCTION) or const (T_CONST) in [directory structure]/kimai2/vendor/symfony/flex/src/Command/RecipesCommand.php on line 35

Any ideas would be much appreciated. In a bit of a panic here.

EDIT: I think it was a PHP version issue. I updated to PHP 8.1 to run the command. It looks like the update went through, but now I'm dealing with a 503 error, which hopefully I get to the bottom of.

**Comment by kevinpapst** [comment]
`PHP Parse error` are always related to an invalid PHP version, as you found out.

Try to fix file permissions, this is cause of over 90% of all problems.
Otherwise check var/log/prod.log for the cause.
